tests/unit/backend/layers/business/test_business.py

```diff
@@ -853,6 +852,44 @@ def test_replace_dataset_in_wrong_status_fail(self):
         self.step_function_provider.start_step_function.assert_not_called()
 
 
+class TestDeleteDataset(BaseBusinessLogicTestCase):
+    def test_delete_dataset_ok(self):
+        collection = self.initialize_unpublished_collection()
+        dataset_version_id_to_remove = collection.datasets[0].version_id
+        dataset_version_id_strs = [dv.version_id.id for dv in collection.datasets]
+        self.assertIn(dataset_version_id_to_remove.id, dataset_version_id_strs)
```

Comment:
Is this check needed since we are pulling the datasets from the collection.datasets?

Reply:
It's a bit pedantic, you're not wrong, but I generally prefer to show that deletion worked by demonstrating it _was removed_, not by demonstrating that it _is not in the collection_.